lib/omnibus/compressors/dmg.rb

```
@@ -209,70 +132,29 @@ def set_volume_icon
+    # Create a compressed dmg.
     #
-    # @return [void]
-    #
-    def prettify_dmg
-      log.info(log_key) { "Making the dmg all pretty and stuff" }
-
-      render_template(resource_path("create_dmg.osascript.erb"),
-                      destination: "#{staging_dir}/create_dmg.osascript",
-                      variables: {
-                        volume_name: volume_name,
-                        pkg_name: packager.package_name,
-                        window_bounds: window_bounds,
-                        pkg_position: pkg_position,
-                      })
-
-      Dir.chdir(staging_dir) do
-        shellout! <<-EOH.gsub(/^ {10}/, "")
-          osascript "#{staging_dir}/create_dmg.osascript"
-        EOH
-      end
-    end
-
-    #
-    # Compress the dmg using hdiutil and zlib. zlib offers better compression
-    # levels than bzip2 (10.4+) or LZFSE (10.11+), but takes longer to compress.
-    # We're willing to trade slightly longer build times for smaller package sizes.
-    #
-    # @return [void]
-    #
-    def compress_dmg
-      log.info(log_key) { "Compressing dmg" }
+    def create_compressed_dmg
+      log.info(log_key) { "Creating compressed dmg" }
 
-      Dir.chdir(staging_dir) do
-        shellout! <<-EOH.gsub(/^ {10}/, "")
-          chmod -Rf go-w "/Volumes/#{volume_name}"
-          sync
-          hdiutil unmount "#{@device}"
-          # Give some time to the system so unmount dmg
-          ATTEMPTS=1
-          until [ $ATTEMPTS -eq 6 ] || hdiutil detach "#{@device}"; do
-            sleep 10
-            echo Attempt number $(( ATTEMPTS++ ))
-          done
-          hdiutil convert \\
-            "#{writable_dmg}" \\
-            -format UDZO \\
-            -imagekey \\
-            zlib-level=9 \\
-            -o "#{package_path}" \\
-            -puppetstrings
-        EOH
-      end
+      shellout! <<-EOH.gsub(/^ {8}/, "")
+        pip install dmgbuild==1.6.5
```

`omnibus` doesn't seem to have proper Python deps: should I move this one somewhere else?

---

I do think that we should try to change this as soon as we can; hiding a python package installation in the middle of an unrelated command looks really bad (additional network dependency + modifying whichever python environment this happens to run on).

This is also already assuming that `pip` is available, so we may as well assume that `dmgbuild` is installed already at this point, and let it fail otherwise.

It doesn't immediately solve the network issue, but I think moving this to https://github.com/DataDog/datadog-agent/blob/71400d2c068e388554898fd6b4f9accf16825df6/.gitlab/package_build/dmg.yml#L2-L5 would make more sense, and give us an opportunity to eventually use the gitlab cache. At least all the pip related work would stay in the same place.

---

Filed DataDog/datadog-agent-dev#137 - let's see...